server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -757,6 +762,12 @@ private synchronized void cleanCache(double stalenessThreshold) {
                         }
                     }
                 }
+
+                if (cleanupKeysFromClosedShards.contains(shardIdInfo)) {
+                    // Since the shard is closed, the cache should drop stats for this shard.
+                    // This should not happen on a full cache cleanup.
```

Comment:
I am guessing we are not dropping stats during cache cleanup as indexShard might still be present on that node, so we still need to count stats for future incoming requests for same shard?
Can we also an IT/UT test around this scenario if not already?

Reply:
Yeah, on cache cleanup we don't drop hits/misses/evictions, to match the behavior of the existing stats API. I suspect this is why they decided to do it that way. I think CacheStatsAPIIndicesRequestCacheIT.testCacheClear() should cover this scenario, right?

Reply:
1. It seems like we should also drop the stats for the case when `cacheEntity==null` on line 756 which represents the shard getting deleted and closed.
2. Currently the clean API seems inefficient where it will call `clear` for each shard on the node. I think outside of this PR lets see if we can optimize it.